tasks/src/main/scala/dagr/tasks/picard/VcfFormatConverter.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package dagr.tasks.picard
+
+import dagr.tasks.DagrDef.PathToVcf
+
+import scala.collection.mutable.ListBuffer
+
+class VcfFormatConverter(val in: PathToVcf, val out: PathToVcf, val requireIndex: Boolean = true)
```

Comment:
Why is require true?

Reply:
It's already the default of the tool but my main reason is that the developer recommends that for best results, an index file is present and `REQUIRED_INDEX` is set to `true`:
```
Converts VCF to BCF or BCF to VCF.  This tool converts files between the
plain-text VCF format and its binary compressed equivalent, BCF. Input and
output formats are determined by file extensions specified in the file names. 
For best results, it is recommended to ensure that an index file is present
and set the REQUIRE_INDEX option to true.

...

REQUIRE_INDEX=Boolean   Fail if an index is not available for the input
						VCF/BCF  Default value: true. This option can be
 						set to 'null' to clear the default value.
						Possible values: {true, false}
``` 

I chose an explicit `true` over an implicitl default `true` for the above reason.

<br>

I could do the following, but since the Picard tool defaults to `true`, you can never turn off the requirement using a `dagr` task defined this way.

```scala
..., requireIndex: Boolean = false ...

if (requireIndex) buffer.append("REQUIRE_INDEX=true")
```

What would you prefer?

Reply:
I really want to avoid having to set an argument like:

```scala
doNotRequireIndex: Boolean = false
```

Reply:
So ... I'm going to go with "that documentation is bs".  `VcfFormatConverter` shouldn't even have that option - it's really very dumb.  The tool _never_ does any index querying, and doesn't touch the index other than to assert that it exists.  I say this both having a) reviewed the code and b) knowing the history of that tool due to it being someone working for me that wrote it back in the day.

I'd suggest we don't even expose the `requireIndex` parameter in dagr and simply hard code it to false in the invocation.

Reply:
@tfenne, great context. Done.